pkg/segment/search/segsearch.go

```diff
@@ -397,28 +397,43 @@ func rawSearchSingleSPQMR(multiReader *segread.MultiColSegmentReader, req *struc
 
 		isBlkFullyEncosed := tRange.AreTimesFullyEnclosed(blkSum.LowTs, blkSum.HighTs)
 		if blkResults.ShouldIterateRecords(aggsHasTimeHt, isBlkFullyEncosed, blkSum.LowTs, blkSum.HighTs, false) {
-			for recNum := uint(0); recNum < numRecsInBlock; recNum++ {
-				if pqmr.DoesRecordMatch(recNum) {
+
+			var recordNums []uint16
+			if req.BlockToValidRecNums != nil {
+				records, ok := req.BlockToValidRecNums[blockNum]
+				if !ok {
+					// TODO: do this check in the caller.
+					return
```

Comment:
Should this be a `continue`, or is `return` correct?

Reply:
We are processing block over here, so if the records for that block are not present we should just return.

Reply:
But this is inside this loop on line 366
```
for blockNum := range filterBlockRequestsChan {
```
So couldn't we be processing multiple blocks and we'd just want to skip the current block? Maybe I'm missing something

Reply:
Oh yeah, sorry missed that.

Reply:
addressed in the latest commit.